Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,903 @@
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+
+            Write-Host "Fetching all Azure Database for PostgreSQL flexible server(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $DBForPostgreSQLFSResources = Import-Csv -LiteralPath $FilePath
+
+            $validDBForPostgreSQLFSResources = $DBForPostgreSQLFSResources | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+
+            $validDBForPostgreSQLFSResources | ForEach-Object {
+                $resourceId = $_.ResourceId
+
+                try {                
+                    $DBForPostgreSQLFSResource = Get-AzPostgreSqlFlexibleServer -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+            
+                    $tlsparameterValue = (Get-AzPostgreSqlFlexibleServerConfiguration -Name $([Constants]::ParameterName_TLS)  -ResourceGroupName $_.ResourceGroupName -ServerName $_.ResourceName -SubscriptionId $SubscriptionId).Value
+                    $sslparameterValue = (Get-AzPostgreSqlFlexibleServerConfiguration -Name $([Constants]::ParameterName_SSL)  -ResourceGroupName $_.ResourceGroupName -ServerName $_.ResourceName -SubscriptionId $SubscriptionId).Value
+                    $DBForPostgreSQLFlexibleServerDetails += $DBForPostgreSQLFSResource  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                    @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                    @{N = 'ResourceName'; E = { $_.Name } }, 
+                    @{N = 'TLSVersion'; E = { $tlsparameterValue } },
+                    @{N = 'SecureTransportStatus'; E = { $sslparameterValue } }
+
+                }
+                catch {
+                    Write-Host "Error fetching Azure Database for PostgreSQL flexible server(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+        }
+                                                                
+    }
+    
+    $totalDBForPostgreSQLFS = ($DBForPostgreSQLFlexibleServerDetails | Measure-Object).Count
+
+    if ($totalDBForPostgreSQLFS -eq 0) {
+        Write-Host "No Azure Database for PostgreSQL flexible server(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+  
+    Write-Host "Found [$($totalDBForPostgreSQLFS)] Azure Database for PostgreSQL flexible server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Azure Database for PostgreSQL flexible server(s) for which server parameter require_secure_transport in not ON
+    $DBForPostgreSQLFSWithNonSecureTLSVersionEnabled = @()
```

Comment:
This is holding resources non-compliant for both require_secure_transport and ssl_min_protocol_version right? Comment and variable are not inline to this

Reply:
Fixed.